reframe/core/modules.py

```diff
@@ -260,6 +260,20 @@ def searchpath_remove(self, *dirs):
         """Remove ``dirs`` from the module system search path."""
         return self._backend.searchpath_remove(*dirs)
 
+    def get_load_string(self, name):
+        """Load string for name."""
+        ret = ''
+        for m in self.resolve_module(name):
+            ret += str(self._backend.get_load_string(m))
+        return ret
+
+    def get_unload_string(self, name):
+        """Unload string for name."""
+        ret = ''
+        for m in self.resolve_module(name):
+            ret += str(self._backend.get_unload_string(m))
+        return ret
```

Comment:
- This is buggy in the case we have a mapping that maps from `m0` to `m1` and `m2`. It will generate a string like `module load m1module load m2`.
- Therefore, conversely to the implementation backends, these two methods here must return a list of commands. Also put an appropriate docstring for the updated functions.
- The call you do in the backend violates the convention that the backend works only with `Module` instances. Here you pass a string.

Following these suggestions, you could rewrite the function as follows:

```python
def emit_load_commands(self, name):
    return [self._backend.emit_load_command(Module(m)) for m in self.resolve_module(m)]
```

Reply:
done